rmf_fleet_adapter/src/lift_supervisor/Node.cpp

```diff
@@ -53,14 +53,49 @@ Node::Node()
 //==============================================================================
 void Node::_adapter_lift_request_update(LiftRequest::UniquePtr msg)
 {
-  // For now we will simply forward the request along
+  if (_log.find(msg->lift_name) == _log.end())
+    _log[msg->lift_name] = "None";
+  
+  std::string& curr_session = _log[msg->lift_name];
+  if (curr_session == "None")
+  {
+    if (msg->request_type != LiftRequest::REQUEST_END_SESSION)
+    {
+      curr_session = msg->session_id;
+      _lift_request_pub->publish(*msg);
```

Comment:
Thanks for the insight. In the latest commit, I use the lift_state callback function to periodically check if the request is complete and re-transmit if needed. I think this can play the role of the timer since lift_state messages are periodically published.